apps/mymedicare_cb/views.py

```diff
@@ -111,22 +127,20 @@ def callback(request):
 
     # Get first and last name from FHIR if not in OIDC Userinfo response.
     if user_info['given_name'] == "" or user_info['family_name'] == "":
-        if 'entry' in response.json():
-            if 'name' in response.json()['entry'][0]['resource']:
-                names = response.json()['entry'][0]['resource']['name']
-                first_name = ""
-                last_name = ""
-                for n in names:
-                    if n['use'] == 'usual':
-                        last_name = n['family']
-                        first_name = n['given'][0]
-                    if last_name or first_name:
-                        user.first_name = first_name
-                        user.last_name = last_name
-                        user.save()
-
-    # Delete the AUS since its not longer needed or valid:
-    aus.delete()
```

Comment:
Sorry, forgot to add an explanatory comment.

I actually think this is a bug waiting to happen. Although this technically doesn't violate idempotency requirements, the fact that repeat requests would fail is a bit concerning. It's too easy for the browser to make a first request, discard it for whatever reason, then make a second request that would fail and cause a surprise error.

Reply:
It looks like if this isn't removed an error will occure if the same request is made again in `mymedicare_login` around 151 when it tries to create a new anonymous user state.

Reply:
It won't. The AnonUserState object is just a mapping between a random state and a next URI, and the state is randomly generated right before creating the user. If it did, though, that would actually be a much bigger bug than what we're causing here.